P2047-alloc-aware-optional.html

```diff
@@ -175,217 +117,254 @@ <h2>Design decisions:</h2>
   type separate from the usual optional. All the complexity of conditional
   triviality is in std::optional, and std::pmr::optional's allocator-aware
   alias target has none of that. </p>
+  
+<p>
+  The first proposal suggested using a type alias where a non-AA pmr::optional
+  aliases std::optional, and AA pmr::optional aliases a new unnamed type. 
+  However, this causes usability issues. Type deduction in cases like :
+<br/><tt>
+<br/>template &lttypename T&gt
+<br/>void foo(pmr::optional&ltT&gt);
+<br/></tt>
+
+<br/>does not work because of the conditional part of the type alias - type 
+deduction needs to know if the type is AA aware before it can deduce what 
+type it needs. 
```

Comment:
I wonder whether this is the precise description of the problem with deduction. It seems to me that the problem is that it's trying to deduce the equivalent of template &lt;class T&gt; void f(metaprog&lt;T&gt;::type);, and T can't be deduced there. T can be deduced for template &lt;class T&gt; void f(plaintemplate&lt;T&gt;);, but not for that template metaprogram argument from which we dig out a nested type.

Reply:
fixed